### DISCUSSION TITLE
The SingleOperandHolder is unnecessary

### DISCUSSION THREAD

**Discussion by ariel-m-s** [encode/django-rest-framework#7914] [Ideas & Suggestions]
Currently, the permissions module defines two `OperandHolder`s, which I think is unnecessary.

https://github.com/encode/django-rest-framework/blob/fd017d00f938c6629d2eeb1b4d81716dff6d006e/rest_framework/permissions.py#L11-L47

The `SingleOperandHolder` is only used for the negation operation (`NOT`). Instead, this could be implemented with a single class that works for all operations. I believe this implementation improves redability.

```python
class OperationHolderMixin:
    ...
    ...

    def __invert__(self):
        return OperandHolder(NOT, self)


# A single class for an arbitrary number of operands.
class OperandHolder(OperationHolderMixin):
    def __init__(self, operator_class, *operand_classes):
        self.operator_class = operator_class
        self.operand_classes = operand_classes

    def __call__(self, *args, **kwargs):
        operands = map(lambda op: op(*args, **kwargs), self.operand_classes)
        return self.operator_class(*operands)
```

**Comment by lovelydinosaur** [comment]
So, at this point in the project lifespan we're *very* much tending towards *not* introducing any code changes unless clearly necessary, since they *always* come with potentially unexpected risks.

I'm not necessarily against this, if the pull request looked super obviously sensible, but. 🤷‍♂️
